packages/core/src/TxBuilders/TxBuilder.Lucid.V1.class.ts

```diff
@@ -246,6 +247,20 @@ export class TxBuilderLucidV1 extends TxBuilder {
       { outputIndex: utxo.index, txHash: utxo.hash },
     ]);
 
+    /**
+     * If we can properly deserialize the order datum using a V3 type, then it's a V3 order.
+     * If not, then we can assume it is a normal V1 order.
+     */
+    try {
+      Data.from(utxoToSpend?.[0]?.datum as string, OrderDatum);
+      console.log("This is a V3 order! Calling appropriate builder...");
+      const v3Builder = new TxBuilderLucidV3(
+        this.lucid,
+        new DatumBuilderLucidV3(this.network)
+      );
+      return v3Builder.cancel({ ...cancelArgs });
+    } catch (e) {}
```

Comment:
Should we handle the catch here in any way? Like treating it as a v1 order?

Reply:
The only reason we are doing a try/catch, is because deserialization will fail if it doesn't conform to the type. The catch is unnecessary in this case, because we will just continue as normal. In V3, the logic is inverse.